runtime/src/bank_forks.rs

```diff
@@ -165,12 +165,11 @@ impl BankForks {
         self.descendants.clone()
     }
 
-    pub fn frozen_banks(&self) -> HashMap<Slot, Arc<Bank>> {
+    pub fn frozen_banks(&self) -> impl Iterator<Item = (Slot, Arc<Bank>)> + '_ {
```

Comment:
I'm tempted to make this just return Banks; a caller can do `bank.slot()` if they need to extract the slot (ie to create a `HashMap`)